realm/src/androidTest/java/io/realm/RealmTest.java

```diff
@@ -2161,4 +2161,34 @@ public void testDistinctInvalidTypes() {
             }
         }
     }
+
+    public void testIsInWriteTransaction() {
+        assertFalse(testRealm.isInWriteTransaction());
+        testRealm.beginTransaction();
+        assertTrue(testRealm.isInWriteTransaction());
+        testRealm.commitTransaction();
+        assertFalse(testRealm.isInWriteTransaction());
+        testRealm.beginTransaction();
+        assertTrue(testRealm.isInWriteTransaction());
+        testRealm.cancelTransaction();
+        assertFalse(testRealm.isInWriteTransaction());
+        testRealm.beginTransaction();
```

Comment:
code below are not testing `isInWriteTransaction()`.

calling `commitTransaction()` should be separate test case and others can be removed since we already have `testNestedTransaction()` and  `testCancelTransaction()`.